xformers/sparse/blocksparse_tensor.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from triton.ops.blocksparse import matmul as blocksparse_matmul
+from triton.ops.blocksparse import softmax as blocksparse_softmax
+
+from xformers.ops import masked_matmul
+
+
+class BlockSparseTensor(torch.Tensor):
```

Comment:
I like the abstraction personally, I think that we'll need to be watertight on the mask description so that users understand well what are the options, but it consolidates the code nicely and makes a lot of sense (don't swap attentions when all you want is changing the mask)